applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -997,6 +1023,12 @@ static void disconnect_peripheral(struct hids_peripheral *per)
 	if (per->cfg_chan_rsp) {
 		submit_forward_error_rsp(per, CONFIG_STATUS_WRITE_FAIL);
 	}
+
+	size_t per_id = per - &peripherals[0];
```

Comment:
Consider defining a simple function for calculating per_id from the per instances (e.g. per_to_per_id). Could it be used also in other places? (If so, i would be great to add a dedicated commit with a small refactor before adding this functionality).

Reply:
Currently this is the only place where we use it (there is another function that finds peripheral by configuration channel recipient and it's used in other places). Can we keep it as is?

Reply:
yes